src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -144,6 +144,8 @@ ValueNode GetValueNodeFromGenericArgument (TypeReference genericArgument)
 				// Technically this should be a new value node type as it's not a System.Type instance representation, but just the generic parameter
 				// That said we only use it to perform the dynamically accessed members checks and for that purpose treating it as System.Type is perfectly valid.
 				return new SystemTypeForGenericParameterValue (inputGenericParameter, _context.Annotations.FlowAnnotations.GetGenericParameterAnnotation (inputGenericParameter));
+			} else if (genericArgument is ArrayType arrayType) {
+				return new ArrayValue (new ConstIntValue (arrayType.Rank));
```

Comment:
This is also not correct. This if will run if I do something like:
```
void TestMethod<T>() { return new List<T[]>(); }
```

It needs to return a node which represents the `T[]` here. Maybe we should rename the method to something like `GetTypeValueNodeFromGenericArgument` or `GetValueNodeForGenericArgumentType`. The returned value node should represent the `typeof(T[])` in this case.

So it should be a SystemTypeValue for `System.Array`.